### PR TITLE
Multi lang

### PR DESCRIPTION
**Pull request by froi**
I've added multiple language support (es and en) to the index creation and the search.

I thought it would be good to do since "Spanglish" is common in PR.

### CODE REVIEW

**Comment by jpadilla** [comment]
@froi does this PR replace #9 ?

File: pages/el-otro.js [hunk 1]
```diff
@@ -39,13 +45,22 @@ function Home(props) {
 
 Home.getInitialProps = async function(context) {
   let query = context.query.q;
+  // set up multi language support
```
Comment (jpadilla) [inline pages/el-otro.js]: @froi maybe we could hide this setup on a lib/util we could reuse. Something like:

```
// ./setup-lunr.js
function setupLunr() {
  stemmerSupport(lunr);
  multi(lunr);
  es(lunr);
  lunr.multiLanguage('en', 'es');
  return lunr;
}

export default setupLunr();
```

```
import lunr from './setup-lunr';
...
```

File: pages/el-otro.js [hunk 2]
```diff
@@ -39,13 +45,22 @@ function Home(props) {
...
+  es(lunr);
+  lunr.multiLanguage('en', 'es');
+  
   if (query) {
-    let normalizedQuery = query.toLowerCase();
-    let results = data.filter((result) =>
-      result.text.includes(normalizedQuery)
-    );
+    const normalizedQuery = unidecode(query.toLowerCase());
+    const idx = lunr.Index.load(elOtroIndex);
```
Comment (jpadilla) [inline pages/el-otro.js]: @froi Can this be done at the module's top level, just once?

**Comment by froi** [comment]
It could

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: pages/el-otro.js (hunk 1 above)
Comment (froi) [reply pages/el-otro.js]: Sure. 

One thing to keep in mind, that setup really depends on the index. If you add another that isn’t multi language that setup is not necessary (English stemmer is used by default) or completely different.

File: pages/el-otro.js (hunk 2 above)
Comment (froi) [reply pages/el-otro.js]: I would imagine so. I’ll test it.

Comment (froi) [reply pages/el-otro.js]: Done.

File: pages/el-otro.js (hunk 1 above)
Comment (froi) [reply pages/el-otro.js]: Done.

**Comment by froi** [comment]
@jpadilla 

> It could. I did them separate to leave the decision up to you. I wasn’t sure if you’d even want the multi language.
> 
> Sent with GitHawk

I take this back. Search is working a bit different when adding multi language. It's up to you but I think this should be tested more to make sure that search makes sense. Up to you though 😃 

Take the term `puta`, which will probably one term people will want to search and is a big part of the controversy.

In PR #9 the search return one page, an exact match in page 74.
image

With the multi language support we three pages: 74, 303, 313

This is the lunr output of the search;

```
[
  {
    "ref": "313",
    "score": 5.668,
    "matchData": {
      "metadata": {
        "put": {
          "text": { "position": [ [ 491, 7 ] ] }
        }
      }
    }
  },
  {
    "ref": "74",
    "score": 4.869,
    "matchData": {
      "metadata": {
        "put": {
          "text": { "position": [ [ 1087, 5 ] ] }
        }
      }
    }
  },
  {
    "ref": "303",
    "score": 4.31,
    "matchData": {
      "metadata": {
        "put": {
          "text": { "position": [ [ 1537, 3 ] ] }
        }
      }
    }
  }
]
```

multi-language-search-puta

**Comment by froi** [comment]
Hola @jpadilla :wave:

Esto lleva abierto mucho y no creo que lo vayamos a seguir trabajando a este punto. Lo estoy cerrando.

**Pull request closed without merging**